src/main/java/com/inf2b/projectcodegeneration/services/TransactionService.java

```diff
@@ -0,0 +1,29 @@
+package com.inf2b.projectcodegeneration.services;
+
+import com.inf2b.projectcodegeneration.jwt.JwtTokenProvider;
+import com.inf2b.projectcodegeneration.models.Transaction;
+import com.inf2b.projectcodegeneration.models.User;
+import com.inf2b.projectcodegeneration.repositories.TransactionRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class TransactionService {
+
+    private final TransactionRepository transactionRepository;
+
+    public TransactionService(TransactionRepository transactionRepository){
+        this.transactionRepository = transactionRepository;
+    }
+
+    public Transaction getAll(){
+        return transactionRepository.getAll().orElseThrow(() -> new RuntimeException("No transactions found"));
+    }
+
+    public Transaction getById(int id){
+        return transactionRepository.getById(id).orElseThrow(() -> new RuntimeException("Transaction not found"));
+    }
+
+    public Transaction commitTransaction(Transaction transaction){
+        return transactionRepository.commitTransaction(transaction);
```

Comment:
```suggestion
        return transactionRepository.save(transaction);
```
In plaats van je eigen methode aanmaken kan je gewoon save gebruiken. Deze zit standaard ingebouwd in de repository